### PR TITLE
Adding Microsoft.NETCore.Native package

### PR DESCRIPTION
**Pull request by SedarG**
Changes:
- updating makefile to produce the static lib for System.Globalization.Native
- add a new package for static and shared lib of System.Globalization.Native. Other native libs will follow.

@jkotas, could you please take a look at these changes
/cc @dotnet/corert-contrib

### CODE REVIEW

File: src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.builds [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  <PropertyGroup>
+    <!-- This property must be set to the same value as $(PackageOutputPath) for the nuspecs and nupkgs to be binplaced to the intended location. -->
+    <OutputPath>$(PackageOutputPath)</OutputPath>
+  </PropertyGroup>
+
+  <!-- We always build the identity/redirection package. However, the platform specific runtime-*.nupkg is built based upon the target OS we are building the product for. -->
+  <ItemGroup>
+    <Project Include="Microsoft.NETCore.Native.pkgproj">
+    </Project>
+    <Project Condition="'$(TargetsLinux)' == 'true' and '$(DistroRid)' == 'debian.8-x64'" Include="debian/Microsoft.NETCore.Native.pkgproj">
```
Comment (jkotas) [inline src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.builds]: We should also add the generic `linux-x64` flavor added recently.

Comment (jkotas) [inline src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.builds]: Look for `'$(DistroRid)' == 'linux-x64'`

**Comment by jkotas** [comment]
@janvorli Could you please take a quick look at the CMakeLists.txt change?

File: .gitignore [hunk 2]
```diff
@@ -24,6 +24,7 @@ build/
 bld/
 [Bb]in/
 [Oo]bj/
+localpkg/
```
Comment (weshaggard) [inline .gitignore]: Why is this needed?

**Comment by weshaggard** [comment]
Why not put these into the Microsoft.NETCore.Runtime package? Are there scenarios we don't want or need these next to the runtime?

File: .gitignore (hunk 2 above)
Comment (jkotas) [inline .gitignore]: This is fixing annoying existing problem: full CoreCLR build will create a bunch of these localpkg directories that will show up in `git status`:

```
D:\coreclr>git status
On branch master
Your branch is up-to-date with 'origin/master'.
Untracked files:
  (use "git add <file>..." to include in what will be committed)

        src/.nuget/Microsoft.NETCore.ILAsm/localpkg/
        src/.nuget/Microsoft.NETCore.ILAsm/win/localpkg/
        src/.nuget/Microsoft.NETCore.ILDAsm/localpkg/
        src/.nuget/Microsoft.NETCore.ILDAsm/win/localpkg/
... bunch more of these ...
```

@weshaggard Is there more appropriate way to fix it?

**Comment by jkotas** [comment]
> Why not put these into the Microsoft.NETCore.Runtime package? 

We need System.Globalization.Native for CoreRT - without the rest of CoreCLR.

File: .gitignore (hunk 2 above)
Comment (weshaggard) [inline .gitignore]: I don't even know where those are coming from. @karajas do you think this is a side-effect of your build tools change?

**Comment by weshaggard** [comment]
>We need System.Globalization.Native for CoreRT - without the rest of CoreCLR.

OK. Is S.Global.Native the only shim we have moved from corefx to coreclr? 

@ericstj this is likely going to require we have another "private" package for all the CoreCLR assets being build from CoreCLR for our engineering changes to get these bits into core-setup.

**Comment by jkotas** [comment]
> OK. Is S.Global.Native the only shim we have moved from corefx to coreclr?

Yes, it is the only one. We have not moved it. S.Global.Native was in coreclr since forever. I would not mind moving to corefx repo instead.

File: .gitignore (hunk 2 above)
Comment (SedarG) [inline .gitignore]: It's somewhat unrelated with this change. build.sh creates localpkg folder under src (for reasons that I didn't dig into). It does so even on the baseline. It's producing an annoying `git status`, and even more annoying when you try to do `git add -A`.

Comment (karajas) [inline .gitignore]: Possibly because of https://github.com/dotnet/buildtools/blob/08573119630248c2ee1b636ceaf3ca8fc8a2ecda/src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/Packaging.targets#L69, this would in the bin dir in corefx and hence not required to be added in gitignore.

**Comment by jkotas** [comment]
OSX build failing with `File not found: '/Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx_prtest/bin/Product/OSX.x64.Checked/System.Globalization.Native.so'`
